tables/courses.rb

```diff
@@ -58,20 +62,233 @@ def migrate_batch(batch)
         level.created_at = old.created_at
         level.updated_at = old.updated_at
 
+        # settings required mission category id, so save once before migrating settings
         skip_saving_unless_valid
+        migrate_settings(old, new)
```

Comment:
better check in case new course is skipped and not saved:
```
if new.persisted?
    migrate_settings(old, new)
    new.save!
end
```

(`skip_saving_unless_valid` will log the error and skip saving the new record when the record is not valid )